functions/gemp_stats/index.py

```diff
@@ -1,20 +1,59 @@
 import json
 import boto3
-from os import getenv
+import pymysql
+import os
+from datetime import datetime
+from dotenv import load_dotenv
 
-def lambda_handler(event, context):
+load_dotenv()
+s3 = boto3.client('s3')
 
-  conn = None
-  try:
-    conn = pymysql.connect(rds_host, user=rds_username, passwd=rds_password, db=db_name, connect_timeout=5)
-  except Exception as e:
-    print("Unable to connect to databse")
+rds_host = os.getenv("RDS_HOST")
+rds_username = os.getenv("RDS_USERNAME")
+rds_password = os.getenv("RDS_PASSWORD")
+db_name = os.getenv("DB_NAME")
+s3_bucket_name = os.getenv("S3_BUCKET_NAME")
 
+def etl(conn, query, s3_file_name):
   with conn.cursor() as cur:
-    cur.execute("SELECT COUNT(*) FROM game_history")
+    cur.execute(query)
     conn.commit()
 
+    rows = []
+    for row in cur:
+      row = map(lambda x: x.isoformat() if isinstance(x, datetime) else x, row)
+      rows.append(list(row))
+
+    # Convert rows to JSON and write to the file
+    with open(s3_file_name, 'w') as outfile:
+      json.dump({ "rows": rows }, outfile)
+
+    # Upload JSON to s3
+    with open(s3_file_name, 'rb') as data:
+      s3.upload_fileobj(data, s3_bucket_name, s3_file_name)
+      # print(file.read())
+
+    return True
+
+def handler(event, context):
+  conn = pymysql.connect(host=rds_host, user=rds_username, passwd=rds_password, db=db_name, connect_timeout=5)
```

Comment:
Let's wrap this in a try-catch. Lambda functions need to always return a status. So in this case we need to return a status of connection failure, rather than 200-ok:
```
conn = None
try:
  conn = pymysql.connect
except Exception as e:
  print("unable to connect to database")
  print(e)
  return(json.dumps("status":"unable to connect to database"))
```